schema_bump_dry_run_scripts/genes/gene_bump_dry_run.py

```diff
@@ -0,0 +1,212 @@
+#!/usr/bin/env python
+import logging
+import os
+from typing import Any, Dict, List, Optional, Tuple
+
+import tiledb as tiledb
+from jinja2 import Template
+
+from cellxgene_schema_cli.cellxgene_schema.env import ONTOLOGY_DIR
+from cellxgene_schema_cli.cellxgene_schema.ontology import SupportedOrganisms
+from schema_bump_dry_run_scripts.common import (
+    BASE_API,
+    fetch_private_collections,
+    fetch_private_dataset,
+    fetch_public_datasets,
+    get_headers,
+)
+from schema_bump_dry_run_scripts.logger import configure_logging
+
+configure_logging()
+logger = logging.getLogger()
+
+ctx = tiledb.default_ctx({"vfs.s3.region": "us-west-2"})
+
+
+def generate_report(data) -> str:
+    report = """## Deprecated Terms in Public Datasets:
+
+{% for collection in deprecated_public %}
+Collection ID: {{ collection }}
+Number of Affected Datasets: {{ deprecated_public[collection].num_datasets }}
+Deprecated Terms: 
+    {{ deprecated_public[collection].deprecated_terms  | join(', ') | wordwrap(78) | replace('\n', '\n    ')}}
+
+{% endfor %}
+## Deprecated Genes in Private Datasets:
+
+{% for collection in open_revisions %}
+Collection ID: {{ collection}}
+{% if open_revisions[collection].revision_of %}
+Note--In A Revision of: {{ open_revisions[collection].revision_of }}
+{% endif %}
+Number of Affected Datasets: {{ open_revisions[collection].num_datasets }}
+Deprecated Terms: 
+    {{ open_revisions[collection].deprecated_terms | join(', ') | wordwrap(78) | replace('\n', '\n    ')}}
+
+{% endfor %}
+## The Following Public Collections Will Not Be Auto-Migrated Due To Having an Open Revision:
+{% for collection in non_auto_migrated %}
+{{ collection }}
+{% endfor %}
+"""
+
+    j2_template = Template(report, trim_blocks=True, lstrip_blocks=True)
+    report = j2_template.render(data)
+    return report
+
+
+def get_genes(dataset: dict, stage: str) -> List[str]:
```

Comment:
this is the logic came up with to be compatible with the upcoming cxg processing change, FYI:

```
    with tiledb.open(s3_path, "r") as var:
        var_df = var.df[:]
        suffix = 0
        while f"name_{suffix}" not in var_df.columns:
            suffix += 1
        index_name = "name_{suffix}"
        stored_genes = var_df[index_name].to_list()
```